lexers/lexer.lua

```diff
@@ -1848,41 +1934,31 @@ function M.detect(filename, line)
 		yaml = 'yaml', yml = 'yaml', --
 		zig = 'zig'
 	}
-	local patterns = {
-		['^#!.+[/ ][gm]?awk'] = 'awk', ['^#!.+[/ ]lua'] = 'lua', ['^#!.+[/ ]octave'] = 'matlab',
-		['^#!.+[/ ]perl'] = 'perl', ['^#!.+[/ ]php'] = 'php', ['^#!.+[/ ]python'] = 'python',
-		['^#!.+[/ ]ruby'] = 'ruby', ['^#!.+[/ ]bash'] = 'bash', ['^#!.+/m?ksh'] = 'bash',
-		['^#!.+/sh'] = 'bash', ['^%s*class%s+%S+%s*<%s*ApplicationController'] = 'rails',
-		['^%s*class%s+%S+%s*<%s*ActionController::Base'] = 'rails',
-		['^%s*class%s+%S+%s*<%s*ActiveRecord::Base'] = 'rails',
-		['^%s*class%s+%S+%s*<%s*ActiveRecord::Migration'] = 'rails', ['^%s*<%?xml%s'] = 'xml',
-		['^#cloud%-config'] = 'yaml'
-	}
 
-	for patt, name in pairs(M.detect_patterns) do if line:find(patt) then return name end end
-	for patt, name in pairs(patterns) do if line:find(patt) then return name end end
-	local name, ext = filename:match("[^/\\]+$")
-	if name then
-		-- remove all suffixes
-		local unchanged
-		while #name > 0 and name ~= unchanged do
-			unchanged = name
-			for _, pattern in ipairs(M.ignored_file_patterns) do
-				name = name:gsub(pattern, "")
-			end
-			local n
-			n, ext = name:match"(.-)%.([^.]+)$"
-			if M.ignored_extensions[ext] then
-				name = n
-				ext = nil
-			end
-		end
-		return M.detect_extensions[name]
-			or extensions[name]
-			or M.detect_extensions[ext]
-			or extensions[ext]
+	local lower = ext and ext:lower()
+	return M.detect_extensions[name]
+		or extensions[name]
+		or M.detect_extensions[ext]
+		or extensions[ext]
+		or M.detect_extensions[lower]
+		or extensions[lower]
```

Comment:
I'm not entirely sure how I feel about this. Filenames are case-sensitive on Linux for example, so some applications might not want to auto-lowercase extensions for fear of false-detections. On the other hand, applications that are on case-insensitive filesystems may simply want to pass a lowercase filename to `M.detect()` to get the desired behavior.

Reply:
I don't like it either but I was tired of opening old .XML files and it not being identified properly